userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -835,13 +833,13 @@ static int32_t load_bpf_file(
 	return res;
 }
 
-static void *perf_event_mmap(struct bpf_engine *handle, int fd, uint32_t *size)
+static void *perf_event_mmap(struct bpf_engine *handle, int fd, uint32_t *size, uint64_t buf_num_pages)
 {
 	int page_size = getpagesize();
-	int ring_size = page_size * BUF_SIZE_PAGES;
+	int ring_size = page_size * buf_num_pages;
 	int header_size = page_size;
 	int total_size = ring_size * 2 + header_size;
-	char buf[SCAP_LASTERR_SIZE];
+	char buf[SCAP_LASTERR_SIZE] = {0};
```

Comment:
For communication I think it would be important to also highlight that the dimension given as parameter is multiplied by 2 just so that you can calculate the correct size in MB when you know the page_size in bytes of your system. Q: Is this consistent that way for modern_bpf @Andreagit97? Would you have other ideas on how to make this the most transparent?